models/beethoven_x/fantom/beethoven_x_fantom_bpt_supply.sql

```diff
@@ -6,133 +6,12 @@
-    premints_1 AS (
-        SELECT
-            poolId AS pool_id,
-            t.token,
-            d.delta,
-            ROW_NUMBER() OVER (PARTITION BY poolId ORDER BY evt_block_time ASC) AS rn
-        FROM {{ source('beethoven_x_fantom', 'Vault_evt_PoolBalanceChanged') }} pb
-        CROSS JOIN UNNEST (pb.deltas) WITH ORDINALITY d(delta, i)
-        CROSS JOIN UNNEST (pb.tokens) WITH ORDINALITY t(token, i)
-        WHERE d.i = t.i 
-        AND BYTEARRAY_SUBSTRING(poolId, 1, 20) = t.token          
-        ORDER BY 1 DESC
-    ),
-
-    -- Select the first row for each pool, which counts as the preminted tokens
-    premints_2 AS (
-        SELECT 
-            * 
-        FROM premints_1
-        WHERE rn = 1
-    ),
-
-    -- Calculate preminted BPTs based on pool type
-    premints AS (
-        SELECT 
-            p.address AS bpt, 
-            CASE WHEN pool_type IN ('linear') THEN CAST('5192296858534827628530496329220095' AS INT256) / POWER(10, 18)
-                WHEN pool_type IN ('stable') THEN CAST(m.delta AS INT256) / POWER(10, 18)
-                ELSE 0
-            END AS preminted_bpts
-        FROM pool_labels p 
-        LEFT JOIN premints_2 m ON p.address = BYTEARRAY_SUBSTRING(m.pool_id, 1, 20)
-    ),
-
-    -- Calculating Joins(mint) and Exits(burn) via Swap
-    joins AS (
-        SELECT 
-            DATE_TRUNC('day', evt_block_time) AS block_date, 
-            tokenOut,
-            pool_type,
-            CASE WHEN pool_type IN ('weighted') 
-            THEN 0
-            ELSE SUM(amountOut / POWER(10, 18)) 
-            END AS ajoins
-        FROM {{ source('beethoven_x_fantom', 'Vault_evt_Swap') }} 
-        LEFT JOIN pool_labels ON BYTEARRAY_SUBSTRING(poolId, 1, 20) = address
-        WHERE tokenOut = BYTEARRAY_SUBSTRING(poolId, 1, 20)       
-        GROUP BY 1, 2, 3
-    ),
-
-    exits AS (
-        SELECT 
-            DATE_TRUNC('day', evt_block_time) AS block_date, 
-            tokenIn,
-            pool_type,
-            CASE WHEN pool_type IN ('weighted') 
-            THEN 0
-            ELSE SUM(amountIn / POWER(10, 18)) 
-            END AS aexits
-        FROM {{ source('beethoven_x_fantom', 'Vault_evt_Swap') }} 
-        LEFT JOIN pool_labels ON BYTEARRAY_SUBSTRING(poolId, 1, 20) = address
-        WHERE tokenIn = BYTEARRAY_SUBSTRING(poolId, 1, 20)        
-        GROUP BY 1, 2, 3
-    ),
-
-    joins_and_exits AS (
-        SELECT 
-            j.block_date, 
-            j.tokenOut AS bpt, 
-            SUM(COALESCE(ajoins, 0) - COALESCE(aexits, 0)) OVER (PARTITION BY j.tokenOut ORDER BY j.block_date ASC) AS adelta
-        FROM joins j
-        FULL OUTER JOIN exits e ON j.block_date = e.block_date AND e.tokenIn = j.tokenOut
-    ),
-
-    calendar AS (
-        SELECT 
-            date_sequence AS day
-        FROM unnest(sequence(date('2021-04-21'), date(now()), interval '1' day)) as t(date_sequence)
+{{ 
+    balancer_v2_compatible_bpt_supply_macro(
```

Comment:
Please check if the unmatched records are acceptable:
https://dune.com/queries/3869082

Reply:
unmatched records are more accurate with this new macro, according to beets' subgraph

Reply:
Then I think this PR is good to go.